force-app/classes/src/matchers/EqualsMatcher.cls

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+@isTest
+public class EqualsMatcher implements ArgumentMatcher {
+  private Object callArgumentToMatch;
+
+  public EqualsMatcher(final Object callArgumentToMatch) {
```

Comment:
suggestion: we might want to offer a greater API than `new EqualsMatcher(value)`.

Something like: `Matchers.eq(value)`,`Matchers.ofType(type)` and `Matchers.jsonEquals(value)`

so that we can use it in concrete customers tests and make our own Matchers private in order to be able to refactor them if needed.

What do you think?

Reply:
I love the idea
That is way better as an API 
And it is not that hard with the current state

Do you want this suggestion to be included in this PR ?
Or we could merge this one and do this in another one ?

Reply:
We can do it in another one yes, I created the task here https://github.com/salesforce/apex-mockery/projects/1#card-86776543